.travis.yml

```diff
@@ -20,7 +20,7 @@ branches:
 
 env:
     global:
-        CXXFLAGS="-O3 -Wall -Wextra -Wno-unused-parameter -Wno-empty-body -Wno-format-security" 
+        CXXFLAGS="-O3 -Wall -Wextra -Wno-unused-parameter -Wno-empty-body -Wno-format-security -DMEDI_NO_OPTIONAL_CONST" 
```

Comment:
The problem is that depending on the actual MPI implementation (so for example MPICH, openmpi etc.) send buffer arguments might be declared as const or not. Unfortunately, there is no (easy) way to get that information automatically. It might be even different between versions of openmpi (on travis it is v1.6.5 and its necessary, locally I have v3.x, and its not)... so for now it is trial and error whether this is necessary or not. Maybe I can find that information somewhere, then I will provide it in the wiki.

Reply:
We found a solution for that. At least for OpenMPI and MPICH2. Still, there is no guarantee that it works for all MPI implementations (especially IntelMPI and PlatformMPI)... However, that should already cover the majority of setups.